### PR TITLE
Add versioning to Routers

### LINKED ISSUE #1065: Add Version to the routers
I can already see where this is going, and there will be confusion around them if we don't store the version.

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Description

Add versioning to the Routers; likely to be very important in the future as the number increases, and especially if we need to redeploy. This adds it to the base contract so that all routers should inherit it (at least all of our routers).

## Type of change

- [ ] Bug fix 
- [X] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [X] The diff is legible and has no extraneous changes
- [X] Complex code has been commented, including external interfaces
- [X] Tests have 100% code coverage
- [X] The base branch is either `main`, or there's a description of how to merge

## Issue Resolution

Closes #1065

### CODE REVIEW

File: pkg/vault/test/.contract-sizes/Router [hunk 1]
```diff
@@ -1,2 +1,2 @@
-Bytecode	23.679
-InitCode	24.652
\ No newline at end of file
+Bytecode	23.999
```
Comment (jubeira) [inline pkg/vault/test/.contract-sizes/Router:1]: :triplemonkas:

**Review by jubeira — APPROVED** [review]
LGTM, thanks @EndymionJkb !

I guess adding one more external method to the routers adds some gas to every call :disappointed: and we're now really close to the limit, but still this will be useful in the future.

File: pkg/vault/test/.contract-sizes/Router (hunk 1 above)
Comment (EndymionJkb) [reply pkg/vault/test/.contract-sizes/Router:1]: I noticed that... but it still fits

File: pkg/pool-weighted/test/WeightedPool.test.ts [hunk 2]
```diff
@@ -28,10 +28,19 @@ import { PoolConfigStructOutput } from '@balancer-labs/v3-solidity-utils/typecha
 import { TokenConfigStruct } from '../typechain-types/@balancer-labs/v3-interfaces/contracts/vault/IVault';
 
 describe('WeightedPool', function () {
-  const POOL_SWAP_FEE = fp(0.01);
+  const FACTORY_VERSION = 'Weighted Factory v1';
+  const POOL_VERSION = 'Weighted Pool v1';
+  const ROUTER_VERSION = 'Router v9';
```
Comment (joaobrunoah) [inline pkg/pool-weighted/test/WeightedPool.test.ts:33]: Nice!

**Review by joaobrunoah — APPROVED** [review]
LGTM!

**Pull request merged**